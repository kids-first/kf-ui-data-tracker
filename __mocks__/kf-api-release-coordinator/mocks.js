import {GET_STUDY_RELEASES, GET_RELEASED_STUDY} from '../../src/state/queries';
import getStudyReleases from './responses/getStudyReleases.json';
import getStudyReleasesEmpty from './responses/getStudyReleasesEmpty.json';
import allReleaseStudies from './responses/allReleaseStudies.json';

export const coordMocks = {
  getStudyReleases: {
    request: {
      query: GET_STUDY_RELEASES,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
      },
    },
    result: getStudyReleases,
  },
  getStudyReleasesError: {
    request: {
      query: GET_STUDY_RELEASES,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
      },
    },
    error: new Error('Failed to fetch releases information'),
  },
  getStudyReleasesEmpty: {
    request: {
      query: GET_STUDY_RELEASES,
      variables: {
        id: 'U3R1ZHlOb2RlOlNEXzhXWDhRUTA2',
      },
    },
    result: getStudyReleasesEmpty,
  },
  allReleaseStudies: {
    request: {
      query: GET_RELEASED_STUDY,
    },
    result: allReleaseStudies,
  },
  allReleaseStudiesError: {
    request: {
      query: GET_RELEASED_STUDY,
    },
    error: new Error('Failed to fetch study releases information'),
  },
};
