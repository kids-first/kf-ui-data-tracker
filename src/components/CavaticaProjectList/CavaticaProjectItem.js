import React, {useState} from 'react';
import PropTypes from 'prop-types';
import {
  List,
  Icon,
  Button,
  Confirm,
  Message,
  Modal,
  Popup,
  Divider,
  Header,
} from 'semantic-ui-react';
import TimeAgo from 'react-timeago';
import {longDate} from '../../common/dateUtils';
import {projectOptions} from '../../common/enums';
import {EditProjectModal} from '../../modals';
import {LinkStudyPopup} from './LinkStudyPopup';

const ProjectAttributes = ({projectNode, disabled}) => (
  <List bulleted horizontal>
    <List.Item
      as={Header}
      sub
      disabled={disabled}
      color={
        projectOptions.find(i => i.key === projectNode.projectType).color ||
        'grey'
      }
    >
      {projectOptions.find(i => i.key === projectNode.projectType).text ||
        'Unknown'}
    </List.Item>
    <List.Item disabled={disabled}>
      Created
      {projectNode.createdBy ? ' by ' + projectNode.createdBy + ' ' : ' '}
      <TimeAgo
        live={false}
        date={projectNode.createdOn}
        title={longDate(projectNode.createdOn)}
      />
    </List.Item>
    {projectNode.workflowType && (
      <List.Item disabled={disabled}>{projectNode.workflowType}</List.Item>
    )}
    <List.Item disabled={disabled}>
      <code>{projectNode.projectId}</code>
    </List.Item>
  </List>
);

const ProjectLink = ({projectNode, disableLink}) => {
  if (disableLink) {
    return <List.Header>{projectNode.name}</List.Header>;
  } else {
    return (
      <List.Header
        as="a"
        target="_blank"
        href={`https://cavatica.sbgenomics.com/u/${projectNode.projectId}`}
      >
        {projectNode.name + ' '}
        <Icon link size="small" name="external" />
      </List.Header>
    );
  }
};

const UnlinkButton = ({unlinkProject, study, projectId}) => (
  <Popup
    trigger={
      <Button
        basic
        negative
        floated="right"
        size="mini"
        icon="unlink"
        content="UNLINK"
        className="ml-10"
        onClick={e => e.stopPropagation()}
      />
    }
    header="Are you sure?"
    content={
      <>
        This will unlink the project from its study. It may always be linked
        back later.
        <Divider />
        <Button
          data-testid="delete-confirm"
          negative
          fluid
          icon={<Icon name="unlink" />}
          content="Unlink"
          onClick={e => {
            e.stopPropagation();
            unlinkProject({
              variables: {
                project: projectId,
                study: study.id,
              },
            });
          }}
        />
      </>
    }
    on="click"
    position="top right"
  />
);

export const ImportVolumeButton = ({importVolumeFiles, projectNode}) => {
  const [isOpen, setIsOpen] = useState(false);
  const [error, setError] = useState();
  const [loading, setLoading] = useState(false);
  // Starts from 'confirm' then progresses to 'success'
  const [state, setState] = useState('confirm');

  const handleSubmit = e => {
    if (state === 'success') {
      setIsOpen(false);
      setState('confirm');
      setError('');
      setLoading(false);
      return;
    }
    setLoading(true);

    e.stopPropagation();
    importVolumeFiles({
      variables: {
        project: projectNode.id,
      },
    })
      .then(resp => {
        if (resp.errors) {
          setError(JSON.stringify(resp.errors));
        }
        setLoading(false);
        setState('success');
      })
      .catch(err => {
        setError(err);
        setLoading(false);
      });
  };

  const directory = new Date().toISOString().slice(0, 10) + '-kf-data-delivery';

  return (
    <>
      <Button
        primary
        className="ml-10"
        icon={<Icon name="cloud download" />}
        content="Import Volume"
        labelPosition="left"
        size="mini"
        onClick={() => setIsOpen(true)}
      />

      <Confirm
        open={isOpen}
        header="Import volume files to Cavatica project"
        onCancel={() => {
          setState('confirm');
          setError('');
          setIsOpen(false);
        }}
        onConfirm={handleSubmit}
        confirmButton={
          loading ? (
            <Button loading disabled content="submitting" />
          ) : state === 'confirm' ? (
            'Proceed with Import'
          ) : (
            'OK'
          )
        }
        cancelButton={state === 'confirm' ? 'Cancel' : null}
        content={
          state === 'confirm' ? (
            <Modal.Content>
              <p>
                You are about to import all files from the connected study
                volume's <b>source/</b> prefix into this delivery project. Any
                users that have access to this project will be able to view
                these files.
              </p>
              <p>Here's what's about to happen:</p>
              <List bulleted>
                <List.Item>
                  A new folder named <b>{directory}</b> will be created in the
                  project
                </List.Item>
                <List.Item>
                  All files inside the <b>source/</b> prefix of the study volume
                  will be imported to this folder.
                </List.Item>
                <List.Item>
                  All files uploaded through the Data Tracker under{' '}
                  <b>source/uploads/</b> will <b>not</b> be imported
                </List.Item>
                <List.Item>
                  If the volume has already been imported to this project today,
                  the previous import will be <b>overwritten</b>
                </List.Item>
              </List>
              {error && <Message error header="Error" content={error} />}
            </Modal.Content>
          ) : (
            <Modal.Content>
              <Message
                success
                icon={<Icon size="mini" name="warning" />}
                header="Import Started"
                content={
                  <>
                    <p>Files may take a couple minutes to copy completely.</p>
                    <p>
                      Please view the{' '}
                      <a
                        as="a"
                        target="_blank"
                        rel="noopener noreferrer"
                        href={`https://cavatica.sbgenomics.com/u/${
                          projectNode.projectId
                        }/files/#q?path=${directory}`}
                      >
                        project's file directory{' '}
                        <Icon link size="small" name="external" />
                      </a>{' '}
                      and review it for any files that may not belong.
                    </p>
                  </>
                }
              />
            </Modal.Content>
          )
        }
      />
    </>
  );
};

const CavaticaProjectItem = ({
  projectNode,
  unlinkProject,
  importVolumeFiles,
  disableLink,
  editable = true,
  showLinkButton,
}) => {
  const [isEditing, setIsEditing] = useState(false);

  if (projectNode.deleted) {
    return (
      <List.Item>
        <List.Content floated="right" verticalAlign="middle">
          {unlinkProject && projectNode.study && (
            <UnlinkButton
              unlinkProject={unlinkProject}
              projectId={projectNode.id}
              study={projectNode.study}
            />
          )}
          {projectNode.projectType === 'DEL' && importVolumeFiles && (
            <ImportVolumeButton
              importVolumeFiles={importVolumeFiles}
              projectNode={projectNode}
            />
          )}
        </List.Content>
        <List.Content>
          <Header floated="right" disabled size="tiny">
            Deleted
          </Header>
          <List.Header as={Header} color="grey" size="tiny">
            {projectNode.name + ' '}
          </List.Header>
          <ProjectAttributes projectNode={projectNode} disabled />
        </List.Content>
      </List.Item>
    );
  } else {
    return (
      <>
        <List.Item>
          <List.Content floated="right">
            {projectNode.projectType === 'DEL' && importVolumeFiles && (
              <ImportVolumeButton
                importVolumeFiles={importVolumeFiles}
                projectNode={projectNode}
              />
            )}
            {editable && (
              <Popup
                trigger={
                  <Button
                    primary
                    basic
                    size="mini"
                    floated="right"
                    icon="pencil"
                    content="EDIT"
                    onClick={() => setIsEditing(true)}
                  />
                }
                inverted
                content="Edit this project"
                on="hover"
                position="top right"
              />
            )}
            {!projectNode.study && showLinkButton && (
              <LinkStudyPopup project={projectNode} />
            )}
            {unlinkProject && projectNode.study && (
              <UnlinkButton
                unlinkProject={unlinkProject}
                projectId={projectNode.id}
                study={projectNode.study}
              />
            )}
          </List.Content>
          <List.Content>
            <ProjectLink projectNode={projectNode} disableLink={disableLink} />
            <ProjectAttributes projectNode={projectNode} />
          </List.Content>
        </List.Item>
        <EditProjectModal
          open={editable && isEditing}
          projectNode={projectNode}
          onCloseDialog={() => setIsEditing(false)}
        />
      </>
    );
  }
};

CavaticaProjectItem.propTypes = {
  /** Project object with study object nested if linked */
  projectNode: PropTypes.object.isRequired,
  /** Action to unlink a project taking project id and study id */
  unlinkProject: PropTypes.func,
  /** If disable the external link to Cavatica on project name */
  disableLink: PropTypes.bool,
  /** Whether or not to allow the edit modal to be spawned */
  editable: PropTypes.bool,
};

export default CavaticaProjectItem;
