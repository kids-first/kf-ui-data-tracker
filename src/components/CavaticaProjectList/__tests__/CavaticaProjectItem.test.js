import React from 'react';
import wait from 'waait';
import {render, fireEvent, act} from 'react-testing-library';
import {MemoryRouter} from 'react-router-dom';
import {MockedProvider} from '@apollo/react-testing';
import {mocks} from '../../../../__mocks__/kf-api-study-creator/mocks';
import CavaticaProjectItem from '../CavaticaProjectItem';

const project = {
  node: {
    id:
      'UHJvamVjdE5vZGU6a29sYm1hbmQvc2Qtd2dwOHJ3M3ctZ2F0ay1oYXBsb3R5cGVjYWxsZXI=',
    createdBy: 'kolbmand',
    createdOn: '2019-08-14T18:29:01+00:00',
    projectId: 'kolbmand/sd-wgp8rw3w-gatk-haplotypecaller',
    projectType: 'DEL',
    name: 'SD_WGP8RW3W gatk-haplotypecaller',
    workflowType: 'bwa_mem',
    study: {
      id: 'U3R1ZHlOb2RlOlNEX1dHUDhSVzNX',
      kfId: 'SD_WGP8RW3W',
      name: null,
      shortName: null,
    },
  },
};

it('renders cavatica project item - show unlink project button', async () => {
  const tree = render(
    <MockedProvider mocks={mocks}>
      <MemoryRouter initialEntries={['/cavatica-projects']}>
        <CavaticaProjectItem
          projectNode={project.node}
          unlinkProject={jest.fn()}
        />
      </MemoryRouter>
    </MockedProvider>,
  );
  expect(tree.container).toMatchSnapshot();

  // Click on the unlink project button
  act(() => {
    fireEvent.click(tree.getByText(/UNLINK/));
  });
  await wait();

  expect(tree.container).toMatchSnapshot();
});

it('renders cavatica project item - show edit button', async () => {
  const tree = render(
    <MockedProvider mocks={mocks}>
      <MemoryRouter initialEntries={['/cavatica-projects']}>
        <CavaticaProjectItem projectNode={project.node} editable={true} />
      </MemoryRouter>
    </MockedProvider>,
  );
  expect(tree.container).toMatchSnapshot();

  // Click on the edit project button
  act(() => {
    fireEvent.click(tree.getByText(/EDIT/));
  });
  await wait();

  expect(tree.container).toMatchSnapshot();
});

it('renders cavatica project item - show unlink project button', () => {
  const tree = render(
    <MemoryRouter initialEntries={['/cavatica-projects']}>
      <CavaticaProjectItem
        projectNode={project.node}
        unlinkProject={jest.fn()}
      />
    </MemoryRouter>,
  );
  expect(tree.container).toMatchSnapshot();
});

it('renders cavatica project item - disable cavatica link', () => {
  const tree = render(
    <MemoryRouter initialEntries={['/cavatica-projects']}>
      <CavaticaProjectItem projectNode={project.node} disableLink={true} />
    </MemoryRouter>,
  );
  expect(tree.container).toMatchSnapshot();
});

it('renders cavatica project item - hide study link', () => {
  const tree = render(
    <MemoryRouter initialEntries={['/cavatica-projects']}>
      <CavaticaProjectItem projectNode={project.node} hideStudy={true} />
    </MemoryRouter>,
  );
  expect(tree.container).toMatchSnapshot();
});

it('renders import volume confirm - button', () => {
  const mock = jest.fn().mockResolvedValue({data: null});
  const tree = render(
    <ImportVolumeButton importVolumeFiles={mock} projectNode={project.node} />,
  );
  expect(tree.container).toMatchSnapshot();
});

it('renders import volume button - modal open', () => {
  const mock = jest.fn().mockResolvedValue({data: null});
  const tree = render(
    <ImportVolumeButton importVolumeFiles={mock} projectNode={project.node} />,
  );
  act(() => {
    fireEvent.click(tree.getByText(/Import Volume/));
  });

  expect(tree.container).toMatchSnapshot();
});

it('renders import volume button - success', async () => {
  const mock = jest.fn().mockResolvedValue({data: null});

  const tree = render(
    <ImportVolumeButton importVolumeFiles={mock} projectNode={project.node} />,
  );

  act(() => {
    fireEvent.click(tree.getByText(/Import Volume/));
  });
  await wait();

  act(() => {
    fireEvent.click(tree.getByText(/Proceed with Import/));
  });
  await wait();

  expect(tree.queryByText(/Import Started/)).toBeDefined();

  fireEvent.click(tree.getByText(/OK/));

  await wait();

  expect(
    tree.queryByText(/Import volume files to Cavatica project/),
  ).toBeNull();
});

it('renders import volume button - error', async () => {
  const mock = jest.fn(() => Promise.reject(null));

  const tree = render(
    <ImportVolumeButton importVolumeFiles={mock} projectNode={project.node} />,
  );

  act(() => {
    fireEvent.click(tree.getByText(/Import Volume/));
  });

  act(() => {
    fireEvent.click(tree.getByText(/Proceed with Import/));
  });

  expect(tree.queryByText(/Error/)).toBeDefined();
  expect(tree.queryByText(/Import Started/)).toBeNull();
});

it('renders import volume button - graphql error', async () => {
  const mock = jest.fn(() => Promise.resolve({errors: ['error']}));

  const tree = render(
    <ImportVolumeButton importVolumeFiles={mock} projectNode={project.node} />,
  );

  act(() => {
    fireEvent.click(tree.getByText(/Import Volume/));
  });

  act(() => {
    fireEvent.click(tree.getByText(/Proceed with Import/));
  });

  expect(tree.queryByText(/Error/)).toBeDefined();
  expect(tree.queryByText(/Import Started/)).toBeNull();
});

it('renders import volume button - cancel', async () => {
  const mock = jest.fn().mockResolvedValue({data: null});

  const tree = render(
    <ImportVolumeButton importVolumeFiles={mock} projectNode={project.node} />,
  );

  act(() => {
    fireEvent.click(tree.getByText(/Import Volume/));
  });

  expect(tree.queryByText(/Proceed with Import/)).toBeDefined();

  act(() => {
    fireEvent.click(tree.getByText(/Cancel/));
  });

  expect(tree.queryByText(/Cancel/)).toBeNull();
});
