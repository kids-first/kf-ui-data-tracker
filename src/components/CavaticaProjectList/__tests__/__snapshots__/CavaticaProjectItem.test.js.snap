// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders cavatica project item - disable cavatica link 1`] = `
<div>
  <div
    class="item"
    role="listitem"
  >
    <div
      class="right floated content"
    >
      <button
        class="ui mini basic primary right floated button"
      >
        <i
          aria-hidden="true"
          class="pencil icon"
        />
        EDIT
      </button>
    </div>
    <i
      aria-hidden="true"
      class="paper plane outline icon"
    />
    <div
      class="content"
    >
      <div
        class="header"
      >
        SD_WGP8RW3W gatk-haplotypecaller
      </div>
      <div
        class="ui bulleted horizontal list"
        role="list"
      >
        <div
          class="item"
          role="listitem"
        >
          Created
           by kolbmand 
          <time
            datetime="2019-08-14T18:29:01.000Z"
            title="14 Aug 2019"
          >
            4 months from now
          </time>
        </div>
        <div
          class="item"
          role="listitem"
        >
          bwa_mem
        </div>
        <div
          class="item"
          role="listitem"
        >
          <code>
            kolbmand/sd-wgp8rw3w-gatk-haplotypecaller
          </code>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders cavatica project item - show edit button 1`] = `
<div>
  <div
    class="item"
    role="listitem"
  >
    <div
      class="right floated content"
    >
      <button
        class="ui mini basic primary right floated button"
      >
        <i
          aria-hidden="true"
          class="pencil icon"
        />
        EDIT
      </button>
    </div>
    <i
      aria-hidden="true"
      class="paper plane outline icon"
    />
    <div
      class="content"
    >
      <a
        class="header"
        href="https://cavatica.sbgenomics.com/u/kolbmand/sd-wgp8rw3w-gatk-haplotypecaller"
        target="_blank"
      >
        SD_WGP8RW3W gatk-haplotypecaller 
        <i
          aria-hidden="true"
          class="external small link icon"
        />
      </a>
      <div
        class="ui bulleted horizontal list"
        role="list"
      >
        <div
          class="item"
          role="listitem"
        >
          Created
           by kolbmand 
          <time
            datetime="2019-08-14T18:29:01.000Z"
            title="14 Aug 2019"
          >
            4 months from now
          </time>
        </div>
        <div
          class="item"
          role="listitem"
        >
          bwa_mem
        </div>
        <div
          class="item"
          role="listitem"
        >
          <code>
            kolbmand/sd-wgp8rw3w-gatk-haplotypecaller
          </code>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders cavatica project item - show edit button 2`] = `
<div>
  <div
    class="item"
    role="listitem"
  >
    <div
      class="right floated content"
    >
      <button
        class="ui mini basic primary right floated button"
      >
        <i
          aria-hidden="true"
          class="pencil icon"
        />
        EDIT
      </button>
    </div>
    <i
      aria-hidden="true"
      class="paper plane outline icon"
    />
    <div
      class="content"
    >
      <a
        class="header"
        href="https://cavatica.sbgenomics.com/u/kolbmand/sd-wgp8rw3w-gatk-haplotypecaller"
        target="_blank"
      >
        SD_WGP8RW3W gatk-haplotypecaller 
        <i
          aria-hidden="true"
          class="external small link icon"
        />
      </a>
      <div
        class="ui bulleted horizontal list"
        role="list"
      >
        <div
          class="item"
          role="listitem"
        >
          Created
           by kolbmand 
          <time
            datetime="2019-08-14T18:29:01.000Z"
            title="14 Aug 2019"
          >
            4 months from now
          </time>
        </div>
        <div
          class="item"
          role="listitem"
        >
          bwa_mem
        </div>
        <div
          class="item"
          role="listitem"
        >
          <code>
            kolbmand/sd-wgp8rw3w-gatk-haplotypecaller
          </code>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders cavatica project item - show unlink project button 1`] = `
<div>
  <div
    class="item"
    role="listitem"
  >
    <div
      class="right floated content"
    >
      <button
        class="ui mini basic primary right floated button"
      >
        <i
          aria-hidden="true"
          class="pencil icon"
        />
        EDIT
      </button>
      <button
        class="ui mini basic negative right floated button ml-10"
      >
        <i
          aria-hidden="true"
          class="unlink icon"
        />
        UNLINK
      </button>
    </div>
    <i
      aria-hidden="true"
      class="paper plane outline icon"
    />
    <div
      class="content"
    >
      <a
        class="header"
        href="https://cavatica.sbgenomics.com/u/kolbmand/sd-wgp8rw3w-gatk-haplotypecaller"
        target="_blank"
      >
        SD_WGP8RW3W gatk-haplotypecaller 
        <i
          aria-hidden="true"
          class="external small link icon"
        />
      </a>
      <div
        class="ui bulleted horizontal list"
        role="list"
      >
        <div
          class="item"
          role="listitem"
        >
          Created
           by kolbmand 
          <time
            datetime="2019-08-14T18:29:01.000Z"
            title="14 Aug 2019"
          >
            4 months from now
          </time>
        </div>
        <div
          class="item"
          role="listitem"
        >
          bwa_mem
        </div>
        <div
          class="item"
          role="listitem"
        >
          <code>
            kolbmand/sd-wgp8rw3w-gatk-haplotypecaller
          </code>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders cavatica project item - show unlink project button 2`] = `
<div>
  <div
    class="item"
    role="listitem"
  >
    <div
      class="right floated content"
    >
      <button
        class="ui mini basic primary right floated button"
      >
        <i
          aria-hidden="true"
          class="pencil icon"
        />
        EDIT
      </button>
      <button
        class="ui mini basic negative right floated button ml-10"
      >
        <i
          aria-hidden="true"
          class="unlink icon"
        />
        UNLINK
      </button>
    </div>
    <i
      aria-hidden="true"
      class="paper plane outline icon"
    />
    <div
      class="content"
    >
      <a
        class="header"
        href="https://cavatica.sbgenomics.com/u/kolbmand/sd-wgp8rw3w-gatk-haplotypecaller"
        target="_blank"
      >
        SD_WGP8RW3W gatk-haplotypecaller 
        <i
          aria-hidden="true"
          class="external small link icon"
        />
      </a>
      <div
        class="ui bulleted horizontal list"
        role="list"
      >
        <div
          class="item"
          role="listitem"
        >
          Created
           by kolbmand 
          <time
            datetime="2019-08-14T18:29:01.000Z"
            title="14 Aug 2019"
          >
            4 months from now
          </time>
        </div>
        <div
          class="item"
          role="listitem"
        >
          bwa_mem
        </div>
        <div
          class="item"
          role="listitem"
        >
          <code>
            kolbmand/sd-wgp8rw3w-gatk-haplotypecaller
          </code>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders cavatica project item - show unlink project button 3`] = `
<div>
  <div
    class="item"
    role="listitem"
  >
    <div
      class="right floated content"
    >
      <button
        class="ui mini basic primary right floated button"
      >
        <i
          aria-hidden="true"
          class="pencil icon"
        />
        EDIT
      </button>
      <button
        class="ui mini basic negative right floated button ml-10"
      >
        <i
          aria-hidden="true"
          class="unlink icon"
        />
        UNLINK
      </button>
    </div>
    <i
      aria-hidden="true"
      class="paper plane outline icon"
    />
    <div
      class="content"
    >
      <a
        class="header"
        href="https://cavatica.sbgenomics.com/u/kolbmand/sd-wgp8rw3w-gatk-haplotypecaller"
        target="_blank"
      >
        SD_WGP8RW3W gatk-haplotypecaller 
        <i
          aria-hidden="true"
          class="external small link icon"
        />
      </a>
      <div
        class="ui bulleted horizontal list"
        role="list"
      >
        <div
          class="item"
          role="listitem"
        >
          Created
           by kolbmand 
          <time
            datetime="2019-08-14T18:29:01.000Z"
            title="14 Aug 2019"
          >
            4 months from now
          </time>
        </div>
        <div
          class="item"
          role="listitem"
        >
          bwa_mem
        </div>
        <div
          class="item"
          role="listitem"
        >
          <code>
            kolbmand/sd-wgp8rw3w-gatk-haplotypecaller
          </code>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders import volume button - modal open 1`] = `
<div>
  <button
    class="ui mini icon primary left labeled button ml-10"
  >
    <i
      aria-hidden="true"
      class="cloud download icon"
    />
    Import Volume
  </button>
</div>
`;

exports[`renders import volume confirm - button 1`] = `
<div>
  <button
    class="ui mini icon primary left labeled button ml-10"
  >
    <i
      aria-hidden="true"
      class="cloud download icon"
    />
    Import Volume
  </button>
</div>
`;
